### PR TITLE
Issue 344 - Re-propose upgrade with same name

### PR DESCRIPTION
**Pull request by abdulla-ashurov**
- Added implementation, when proposing a new upgrade, if there is an expired upgrade proposal with the same name, we remove the proposal and add a new one (with the same name and all votes cleared) instead of failing with "Proposed upgrade with name=%v already exists on the ledger".

### CODE REVIEW

File: x/dclupgrade/keeper/msg_server_propose_upgrade.go [hunk 1]
```diff
@@ -45,6 +33,20 @@ func (k msgServer) ProposeUpgrade(goCtx context.Context, msg *types.MsgProposeUp
 		return nil, err
 	}
 
+	// check if proposed upgrade with the same name already exists and propose plan height
+	existProposedUpgrade, isFound := k.GetProposedUpgrade(ctx, msg.Plan.Name)
```
Comment (uabjabborov) [inline x/dclupgrade/keeper/msg_server_propose_upgrade.go:37]: Better name would be `existingProposedUpgrade`

File: x/dclupgrade/keeper/msg_server_approve_upgrade.go [hunk 2]
```diff
@@ -69,6 +69,11 @@ func (k msgServer) ApproveUpgrade(goCtx context.Context, msg *types.MsgApproveUp
 		approvedUpgrage := types.ApprovedUpgrade(proposedUpgrade)
 		k.SetApprovedUpgrade(ctx, approvedUpgrage)
 	} else {
+		if ctx.BlockHeight() > proposedUpgrade.Plan.Height {
+			return nil, sdkerrors.Wrapf(sdkerrors.ErrUnauthorized,
+				"upgrade cannot be scheduled in the past",
```
Comment (ashcherbakov) [inline x/dclupgrade/keeper/msg_server_approve_upgrade.go:74]:
```suggestion
				"upgrade cannot be scheduled in the past. Consider re-proposing upgrade with the same name and a new block height from the future.",
```

File: x/dclupgrade/keeper/msg_server_propose_upgrade.go (hunk 1 above)
Comment (abdulla-ashurov) [reply x/dclupgrade/keeper/msg_server_propose_upgrade.go:37]: Done! ✅

File: x/dclupgrade/keeper/msg_server_propose_upgrade.go [hunk 3]
```diff
@@ -45,6 +33,20 @@ func (k msgServer) ProposeUpgrade(goCtx context.Context, msg *types.MsgProposeUp
...
+	if isFound && ctx.BlockHeight() > existProposedUpgrade.Plan.Height && ctx.BlockHeight() < msg.Plan.Height {
```
[the remark this answers is not in the document]
Comment (abdulla-ashurov) [reply x/dclupgrade/keeper/msg_server_propose_upgrade.go]: Done! ✅

File: x/dclupgrade/keeper/msg_server_approve_upgrade.go [hunk 4]
```diff
@@ -69,6 +69,11 @@ func (k msgServer) ApproveUpgrade(goCtx context.Context, msg *types.MsgApproveUp
...
+			)
+		}
```
[the remark this answers is not in the document]
Comment (abdulla-ashurov) [reply x/dclupgrade/keeper/msg_server_approve_upgrade.go]: Done! ✅

File: x/dclupgrade/keeper/msg_server_approve_upgrade.go (hunk 2 above)
Comment (abdulla-ashurov) [reply x/dclupgrade/keeper/msg_server_approve_upgrade.go]: Done!

**Review by ashcherbakov — APPROVED** [review]

**Pull request merged**